lib/core/connector_settings.rb

```diff
@@ -114,6 +114,16 @@ def valid_index_name?
       index_name&.start_with?(Utility::Constants::CONTENT_INDEX_PREFIX)
     end
 
+    def ready_for_sync?
+      Connectors::REGISTRY.registered?(service_type) &&
+        valid_index_name? &&
+        connector_status_allows_sync?
+    end
+
+    def running?
+      @elasticsearch_response[:_source][:last_sync_status] == Connectors::SyncStatus::IN_PROGRESS
```

Comment:
Might be worth to add some sort of a killswitch to also not consider jobs that are stuck/orphaned, though probably not important for now.

Reply:
I would not change it until(if) we stop relying on connectors document to determine the job execution state.